src/main/java/net/fabricmc/loader/launch/knot/KnotClassLoader.java

```diff
@@ -146,10 +148,31 @@ public URL nextElement() {
 
 					int pkgDelimiterPos = name.lastIndexOf('.');
 					if (pkgDelimiterPos > 0) {
-						// TODO: package definition stub
-						String pkgString = name.substring(0, pkgDelimiterPos);
-						if (getPackage(pkgString) == null) {
-							definePackage(pkgString, null, null, null, null, null, null, null);
+
+						String pkgName = name.substring(0, pkgDelimiterPos);
+						Package currentPackage = getPackage(pkgName);
+
+						if (currentPackage != null) {
+							if (currentPackage.isSealed()) {
+								if (!currentPackage.isSealed(metadata.codeSource.getLocation())) {
+									KnotClassDelegate.Metadata realMeta = delegate.packages.get(currentPackage);
+									URL source = realMeta == null ? null : realMeta.codeSource.getLocation();
+									throw new SecurityException("Cannot load the class " + name + " from "
+										+ ModResolver.describeRealLocation(metadata.codeSource.getLocation())
+										+ " because it's package has already been loaded (and sealed) from "
```

Comment:
```suggestion
										+ " because its package has already been loaded (and sealed) from "
```

Applies to all other instances of "it's" in this PR.

Reply:
Ok, was that all?